### ISSUE TITLE
Test issues without test issue keys not being created as expected

### ISSUE THREAD

**Issue by barnes** [Qytera-Gmbh/cypress-xray-plugin#47]
Looking through the documentation, it seems the expectation is that a new test issue will be created for any tests not found in Jira, but that is not currently happening in my case. 

Config file e2e: 

```js
import { addXrayResultUpload, configureXrayPlugin } from "cypress-xray-plugin/plugin";

...

 e2e: {
    async setupNodeEvents(on, config) {
      await configureXrayPlugin({
        jira: {
          projectKey: 'QTP'
        },
        xray: {
          uploadResults: true
        },
      });
      await addXrayResultUpload(on);
    },
  }, 
```

Example of CLI warning: 

``` 
| Cypress Xray Plugin | WARNING | No test issue key found for test "test - general login validation tests tests with good email, bad pass". Skipping result upload for this test.
```

If I create a test issue in Jira and prefix the test name in Cypress, the upload functions normally. Screenshot and video uploading working as well.

**Comment by csvtuda** [comment]
Hey @barnes,

thanks for raising this issue. I will try to see if I can replicate it. The behaviour might have changed in the latest major release.

Out of curiosity, what would your desired behaviour be? Should the plugin create issues or should it only work with existing issues?

**Comment by barnes** [comment]
Given I have a large existing test set, I'd prefer it create the cases. But
I can see the argument to prevent that as well. Seems like a decent config
option to include in the future.

I'm thinking it might also have something to do with my Xray settings in
Jira, I'll take a look in the morning.

**Comment by csvtuda** [comment]
No you are right, it's a plugin issue, I've changed this behaviour more or less by accident in version `3.0.0`. I will add an option to configure this behaviour.

**Comment by barnes** [comment]
Confirming that rolling back to 2.0.3 did resolve the issue. Thanks for the quick feedback. I rolled my own Cypress/XRay integration in my previous position and was not looking forward to doing it all over again, but with this it took no effort. Amazing work really.

**Comment by csvtuda** [comment]
Hey @barnes,

I've just published [`3.1.0`](https://www.npmjs.com/package/cypress-xray-plugin). This version should now default to creating test issues again. I've also added the option to configure this behaviour, as described [here](https://qytera-gmbh.github.io/projects/cypress-xray-plugin/section/configuration/jira/#createtestissues).

Feel free to give me a quick feedback/reopen the issue if necessary once you've tried it out.

**Comment by barnes** [comment]
Test Info is not being passed to the JSON with only the createTestIssue flag set to true, so Xray is rejecting. Enabling the overwriteIssueSummary flag as well resolves this. I'm seeing where to resolve this and might have some time this afternoon to fix it and put in a PR.

**Comment by csvtuda** [comment]
Thanks for the feedback, I'd be happy to accept a pull request. I've found at least one more issue, so a bugfix version should be due soon anyways 😞 I am reopening this issue.

Regarding the dash character, I've opened a new issue for this, where I will follow up with updates if available: #49

**Comment by csvtuda** [comment]
I have included another carefully crafted attempt to fix this issue in [version `3.2.0`](https://www.npmjs.com/package/cypress-xray-plugin). Test issues for entirely new test cases should now automatically be created again *without* having to enable `plugin.overWriteIssueSummaries` (as long as `jira.createTestIssues` is enabled, of course).

Once you find the time to try it out as you initially intended, make sure to send some feedback my way.
